### PR TITLE
More complex deployments

### PR DESCRIPTION
**Pull request by SCdF**
Updated to use deploy docs

medic/medic-webapp#3993

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Announced in Changes.md in plain English. Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

### CODE REVIEW

File: api/controllers/upgrade.js [hunk 1]
```diff
@@ -1,68 +1,23 @@
...
-            user: username,
-            version: buildInfo.version,
+const auth = require('../auth'),
+      serverUtils = require('../server-utils');
+
+const service = require('../services/upgrade');
+
+module.exports = {
+  upgrade: (req, res) => {
+    return auth.check(req, '_admin')
```
Comment (garethbowen) [inline api/controllers/upgrade.js]: I think this is inconsistent with the UI - non-db-admins with the "can_configure" permission will be able to see the upgrade button but get an error when they click it. Either add a new permission for "can_update_app" (or a decent name) and use that in the front end too, or change this to use "can_configure".

File: Gruntfile.js [hunk 2]
```diff
@@ -287,6 +287,17 @@ module.exports = function(grunt) {
...
+        cmd: () => `
+          mkdir -p ddocs/medic/build_info;
+          cd ddocs/medic/build_info;
+          echo "medic" > application;
+          echo "medic" > namespace;
+          echo "${releaseName}" > version;
+          echo "${new Date().toISOString()}" > time;
+          echo "grunt on \`whoami\`" > author;
+          echo '["medic-api-0.1.0.tgz", "medic-sentinel-0.1.0.tgz"]' > node_modules.json`
+      },
```
Comment (garethbowen) [inline Gruntfile.js]: I think it's nicer to commit the node_modules.json, application, and namespace files rather than generating them every time.

Comment (SCdF) [reply Gruntfile.js]: Good idea, done!

File: api/controllers/upgrade.js (hunk 1 above)
Comment (SCdF) [reply api/controllers/upgrade.js]: Yeah good point. I think I'll change it to `can_configure` for now… I think that makes sense, in the future we can always restrict it further.

**Pull request merged**